pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/juniper/StatusKitRoute.java

```diff
@@ -0,0 +1,51 @@
+package org.broadinstitute.dsm.route.juniper;
+
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.dsm.exception.DSMBadRequestException;
+import org.broadinstitute.dsm.model.nonpepperkit.KitResponseError;
+import org.broadinstitute.dsm.model.nonpepperkit.NonPepperStatusKitService;
+import org.broadinstitute.dsm.statics.RequestParameter;
+import org.broadinstitute.dsm.statics.RoutePath;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+@Slf4j
+public class StatusKitRoute implements Route {
+    NonPepperStatusKitService nonPepperStatusKitService;
+
+    public StatusKitRoute(NonPepperStatusKitService nonPepperStatusKitService) {
+        this.nonPepperStatusKitService = nonPepperStatusKitService;
+    }
+
+    @Override
+    public Object handle(Request request, Response response) throws Exception {
+
+        if (request.url().contains(RoutePath.STATUS_KIT_STUDY)) {
```

Comment:
It does not look like we check user access at any point?

Reply:
Same as all of the other non UI  endpoints, this is getting passed by authenticating their secret instead of their access roles

Reply:
Yes no special roles are defined for this in database, and also not for Mercury or BSP. We do check the token signature and expiration time and the secret matching the one we have on file.
I think it is, not sure what long term means anymore for DSM, but yes I don't think there is any ticket or talk to change that. If you want we can discuss this more and make plans